plugins/nemo-agents/tests/unit/test_spec_render.py

```
@@ -0,0 +1,195 @@
+        assert round_tripped.tools == tools_md
+
+
+class TestParseErrors:
+    def test_missing_front_matter(self) -> None:
+        with pytest.raises(SpecRenderError, match="front matter"):
+            parse_spec("# Agent Spec\n\n## Job\n\nsomething\n")
+
+    def test_missing_required_section(self) -> None:
+        spec = _minimal_spec()
+        md = render_spec(spec).replace("## Audience\n\ninternal employees\n", "")
+        with pytest.raises(SpecRenderError, match="missing section.*Audience"):
```

Ruff (0.15.14)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)